test/test_main.cpp

```diff
@@ -0,0 +1,29 @@
+#define CATCH_CONFIG_MAIN
+#define CATCH_CONFIG_ENABLE_BENCHMARKING
+#include "catch2/catch.hpp"
+
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+#include <Kokkos_Core.hpp>
+#endif
+
+class testRunListener : public Catch::TestEventListenerBase
+{
+public:
+    using Catch::TestEventListenerBase::TestEventListenerBase;
+
+    void testRunStarting(Catch::TestRunInfo const &) override
+    {
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::initialize();
+#endif
+    }
+
+    void testRunEnded(Catch::TestRunStats const &) override
+    {
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::finalize();
+#endif
+    }
+};
+
+CATCH_REGISTER_LISTENER(testRunListener)
```

Comment:
🤷  It's what I ended up with when I added Kokkos tests to Singe, but that was a few years ago.  I wouldn't be surprised if the change from Catch2 v2 to v3 means both our approaches are wrong.  If you prefer I change things, I can do that.  This was just some monkey-see-monkey-do theft from Singe because that seemed the easiest way to get the ball rolling.

Reply:
Eh if it works it works. I don't feel the need to change it. Was just curious.

Reply:
I went ahead and moved to the singularity-eos version because it worked with zero thought when I migrated to catch2 v3